src/vision/calib/MonoCalibrator.hpp

```diff
@@ -0,0 +1,137 @@
+    {
+        out << "CAPTURING";
+    }
+    else
+    {
+        out << "CALIBRATED";
+    }
+    return out;
+}*/
+
+class Settings
+{
+    public:
+        Settings() : goodInput(false) {}
+        enum Pattern
+        {
+            NON_EXISTANT,
+            CHESSBOARD,
+            CIRCLES_GRID,
+            ASYMMETRIC_CIRCLES_GRID
+        };
+
+        void write(FileStorage& fs) const;
+        void read(const FileNode& node);
+
+        void validate();
+
+        static bool readStringList(const string& filename, vector<string>& l);
+
+    //private:
+        Size boardSize;
+        Pattern calibrationPattern;
+        float squareSize;
+        int nrFrames;
+        float aspectRatio;
+        int delay;
+        bool writePoints;
+        bool writeExtrinsics;
+        bool calibZeroTangentDist;
+        bool calibFixPrincipalPoint;
+        bool flipVertical;
+        string outputFileName;
+        bool showUndistorted;
+        string input;
+        bool useFisheye;
+        bool fixK1;
+        bool fixK2;
+        bool fixK3;
+        bool fixK4;
+        bool fixK5;
+
+        bool goodInput;
+        int flag;
+    private:
+        string patternToUse;
+};
+
+static inline void read(const FileNode& node, Settings& x,
+                        const Settings& default_value = Settings())
+{
+    if (node.empty())
+    {
+        x = default_value;
+    }
+    else
+    {
+        x.read(node);
+    }
+}
+
+static inline void write(FileStorage& fs, const Settings& s)
+{
+    s.write(fs);
+}
+
+bool runCalibration(Settings& s, Size& imageSize, Mat& cameraMatrix,
+                    Mat& distCoeffs, vector<vector<Point2f>> imagePoints,
+                    vector<Mat>& rvecs, vector<Mat>& tvecs,
+                    vector<float>& reprojErrs, double& totalAvgErr);
+
+bool runCalibrationAndSave(Settings& s, Size imageSize, Mat& cameraMatrix,
+                           Mat& distCoeffs,
+                           vector<vector<Point2f>> imagePoints);
+
+void saveCameraParams(Settings& s, Size& imageSize, Mat& cameraMatrix,
+                      Mat& distCoeffs, const vector<Mat>& rvecs,
+                      const vector<Mat>& tvecs, const vector<float>& reprojErrs,
+                      const vector<vector<Point2f>>& imagePoints,
+                      double totalAvgErr);
+
+void calcBoardCornerPositions(Size boardSize, float squareSize,
+                              vector<Point3f>& corners,
+                              Settings::Pattern patternType);
+
+double computeReprojectionErrors(const vector<vector<Point3f>>& objectPoints,
+                                 const vector<vector<Point2f>>& imagePoints,
+                                 const vector<Mat>& rvecs,
+                                 const vector<Mat>& tvecs,
+                                 const Mat& cameraMatrix,
+                                 const Mat& distCoeffs,
+                                 vector<float>& perViewErrors, bool fisheye);
```

Comment:
Separate the settings file class and the function prototypes into two different header files. They are logically two separate things if we ever intend to use the settings class elsewhere.